action.yml

```diff
@@ -39,28 +40,67 @@ runs:
         registry: ghcr.io
         username: ${{ inputs.github_username }}
         password: ${{ inputs.github_token }}
-    - name: Check that the tag exists
-      run: docker manifest inspect ghcr.io/${{ inputs.image }} > /dev/null
+    - name: Prepend ghcr.io to image name if necessary
+      id: image_name
+      uses: actions/github-script@v6
+      env:
+        IMAGE: ${{ inputs.image }}
+      with:
+        script: |
+          image = process.env.IMAGE
+          if(!image.startsWith('ghcr.io/')) {
+            image = "ghcr.io/" + image
+          }
+          core.setOutput('image',image)
+    - name: Check that the image & tag exist
+      run: docker manifest inspect "$IMAGE" > /dev/null
       shell: bash
+      env:
+        IMAGE: ${{ steps.image_name.outputs.image }}
     - uses: azure/setup-kubectl@v1
     - name: Authenticate with kubernetes
       shell: bash
       run: |
+        set -x
         mkdir -p ${HOME}/.kube/certs/cluster
         echo ${{ inputs.cluster_ca }} | base64 -d > ${HOME}/.kube/certs/cluster/k8s-ca.crt
-        kubectl config set-cluster cluster --certificate-authority=${HOME}/.kube/certs/cluster/k8s-ca.crt --server=${{ inputs.cluster_server }}
-        kubectl config set-credentials default --token=`echo ${{ inputs.namespace_token }} | base64 -d`
-        kubectl config set-context default --cluster=cluster --user=default --namespace=${{ inputs.namespace }}
+        kubectl config set-cluster cluster --certificate-authority=${HOME}/.kube/certs/cluster/k8s-ca.crt --server="$CLUSTER_SERVER"
+        kubectl config set-credentials default --token=$(echo "$NAMESPACE_TOKEN" | base64 -d)
+        kubectl config set-context default --cluster=cluster --user=default --namespace="$NAMESPACE"
         kubectl config use-context default
+      env:
+        CLUSTER_CA: ${{ inputs.cluster_ca }}
+        CLUSTER_SERVER: ${{ inputs.cluster_server }}
+        NAMESPACE_TOKEN: ${{ inputs.namespace_token }}
+        NAMESPACE: ${{ inputs.namespace }}
```

Comment:
why the explicitly defined env vars per step?

Reply:
This reduces the opportunities for strange quoting issues -- by putting them in the environment and then using them quoted in the commands, the shell will deal correctly with any whitespace or embedded metacharacters.

Reply:
While I don't think this (https://docs.github.com/en/enterprise-cloud@latest/actions/security-guides/security-hardening-for-github-actions?learn=adopting_github_actions_for_your_enterprise&learnProduct=admin#understanding-the-risk-of-script-injections)  applies directly here as we aren't running in an untrusted context, I think it makes the code more readable and robust